SpatialGDK/Source/SpatialGDK/Public/Utils/SpatialDebugger.h

```diff
@@ -258,4 +296,27 @@ class SPATIALGDK_API ASpatialDebugger : public AInfo
 	FCanvasIcon Icons[ICON_MAX];
 
 	USpatialDebuggerConfigUI* ConfigUIWidget;
+
+	// Actors selected by user for debugging
+	TArray<TWeakObjectPtr<AActor>> SelectedActors;
+
+	// Highlighted actor under the mouse cursor
+	TWeakObjectPtr<AActor> HoverActor;
+	// Highlighted actor original materials and components
+	TArray<UMaterialInterface*> ActorMeshMaterials;
```

Comment:
For proper memory management, these two arrays should be UPROPERTIES, or arrays of weak object ptrs.

Reply:
Updated these to arrays of weak pointers and added IsValid checks in code.